sitemap_utils.py

```
@@ -0,0 +1,367 @@
+"""
+Utility module for parsing XML sitemaps to map website structure before scraping.
+"""
+import logging
+import time
+import re
+from typing import List, Dict, Any, Optional, Set, Tuple
```

Ruff (0.8.2)

`typing.List` is deprecated, use `list` instead

(UP035)